aten/src/ATen/native/cuda/Blas.cu

```diff
@@ -0,0 +1,59 @@
+#include <ATen/ATen.h>
+#include <ATen/Dispatch.h>
+#include <ATen/native/Blas.h>
+#include <ATen/cuda/CUDABlas.h>
+
+namespace at { namespace native {
+namespace {
+
+void addmv_impl_cuda(Tensor& result, const Tensor &self, const Tensor &mat, const Tensor &vec, Scalar beta_, Scalar alpha_) {
+  auto r_stride = result.stride(0);
+  auto vec_size = vec.size(0);
+  auto vec_stride = vec.stride(0);
+
+  if (mat.scalar_type() == kHalf || mat.scalar_type() == kBFloat16) {
```

Comment:
I don't know why in the legacy code `bfloat16` is turned on, but this is never working and never tested. Trying to run

```python
import torch

def run(dtype):
    a = torch.ones((2,), device='cuda', dtype=dtype)
    b = torch.empty((2, 2), device='cuda', dtype=dtype)
    c = torch.empty((2,), device='cuda', dtype=dtype)
    print(a.addmv(b, c, alpha=6, beta=3))

run(torch.float)
run(torch.half)
run(torch.bfloat16)
```

would lead to

```
Traceback (most recent call last):
  File "cublas.py", line 11, in <module>
    run(torch.bfloat16)
  File "cublas.py", line 7, in run
    print(a.addmv(b, c, alpha=6, beta=3))
RuntimeError: Bgemm not supported on at::BFloat16 type
```

on nightly build